### PR TITLE
[task] Implementated Unit testing for POST request positive test case refers #12

### PR DESCRIPTION
**Pull request by sand1p**

### CODE REVIEW

File: test/controllers/SecretControllerSpecGetAPITest.scala [hunk 1]
```diff
@@ -0,0 +1,27 @@
+package controllers
```
Comment (gopinath-langote) [inline test/controllers/SecretControllerSpecGetAPITest.scala]: `/SecretControllerSpecGetAPITest.scala ` Avoid using such test class names.

Writting test class for each endpoint is not good idea.

Class test name can be `/SecretControllerTest.scala`

File: test/controllers/SecretControllerSpecGetAPITest.scala [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+import org.scalatestplus.play._
+import org.scalatestplus.play.guice._
+import play.api.test.Helpers._
+import play.api.test._
+import stubs.StubSecretService
+
+class SecretControllerSpecGetAPITest extends PlaySpec with GuiceOneAppPerTest {
+
+  def validGetSecreteControllerCallReturnsPassword() = {
```
Comment (gopinath-langote) [inline test/controllers/SecretControllerSpecGetAPITest.scala]: `shouldReturnPasswordForASecrete` always use agreement like statements for the test method names.

Practice it more

File: test/controllers/SecretcontrollerSpecPostAPITest.scala [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+import play.api.libs.json.Json
+
+class SecretcontrollerSpecPostAPITest extends PlaySpec with GuiceOneAppPerTest {
+  implicit lazy val materializer: Materializer = app.materializer
+
+  def validPostSecretControllerRequestReturnsValidLink() = {
+    "controllers.SecretController.save POST" should {
+      "return valid link for password" in {
+
+        val secreteController = new SecretController(stubControllerComponents(), new StubSecretService)
```
Comment (gopinath-langote) [inline test/controllers/SecretcontrollerSpecPostAPITest.scala]: Good test!

**Review by gopinath-langote — CHANGES_REQUESTED** [review]

File: test/controllers/SecretControllerSpecGetAPITest.scala (hunk 1 above)
Comment (sand1p) [reply test/controllers/SecretControllerSpecGetAPITest.scala]: We need to cover  positive as well as negative test cases as well, right ?

Comment (gopinath-langote) [reply test/controllers/SecretControllerSpecGetAPITest.scala]: yes

File: test/controllers/SecretControllerTest.scala [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+import java.util.Date
+
+import org.scalatestplus.play._
+import org.scalatestplus.play.guice._
+import play.api.libs.json.Json
+import play.api.test.Helpers._
+import play.api.test._
+import stubs.StubSecretService
+
+class SecretControllerTest extends PlaySpec with GuiceOneAppPerTest {
```
Comment (gopinath-langote) [inline test/controllers/SecretControllerTest.scala:12]: I am interested in looking at the output show on console after runnin test - that will help in naming it

File: test/controllers/SecretControllerTest.scala [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+             |   "expireAt" : $date
+             | }
+             """.stripMargin
+
+        val response = secreteController.save().apply(FakeRequest(POST, "/secrete/").withBody(Json.parse(jsonBody)))
+        status(response) mustBe OK
+      }
+    }
+  }
+  shouldReturnPasswordForValidLink()
```
Comment (gopinath-langote) [inline test/controllers/SecretControllerTest.scala:46]: Why to call def?

Comment (gopinath-langote) [reply test/controllers/SecretControllerTest.scala:46]: Are you following the doc of test lib?

**Review by gopinath-langote — APPROVED** [review]

File: test/controllers/SecretControllerSpecGetAPITest.scala (hunk 2 above)
Comment (sand1p) [reply test/controllers/SecretControllerSpecGetAPITest.scala]: sure

**Pull request merged**